### DISCUSSION TITLE
error code 139

### DISCUSSION THREAD

**Discussion by smo85000** [tillsteinbach/VWsFriend#106] [Q&A]
Hi,
I tried to install VWsFriend on raspberry pi zero w.
But I had a error code when creating the container with docker-compose --env-file ./myconfig.env up
vwsfriend_vwsfriend_1 exited with code 139

can you plesae help me ?

Thanks

Stephane

**Comment by tillsteinbach** [comment]
Hi Stephane, can you provide the full log output?

**Reply by smo85000** [reply]
Thanks for your quick answer ! you can find the output from the command line :

pi@raspVW:~/VWsFriend $ sudo docker-compose --env-file ./myconfig.env up
Creating network "vwsfriend_backend" with the default driver
Creating network "vwsfriend_bridge" with the default driver
Creating vwsfriend_postgresdb_1 ... done
Creating vwsfriend_grafana_1    ... done
Creating vwsfriend_vwsfriend_1  ... done
Attaching to vwsfriend_postgresdb_1, vwsfriend_grafana_1, vwsfriend_vwsfriend_1
postgresdb_1  |
postgresdb_1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgresdb_1  |
postgresdb_1  | 2021-11-23 19:44:53.172 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on arm-unknown-linux-gnueabi, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 32-bit
postgresdb_1  | 2021-11-23 19:44:53.193 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgresdb_1  | 2021-11-23 19:44:53.196 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgresdb_1  | 2021-11-23 19:44:53.277 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgresdb_1  | 2021-11-23 19:44:53.406 UTC [14] LOG:  database system was shut down at 2021-11-23 19:41:43 UTC
postgresdb_1  | 2021-11-23 19:44:53.681 UTC [1] LOG:  database system is ready to accept connections
vwsfriend_grafana_1 exited with code 139
vwsfriend_vwsfriend_1 exited with code 139

**Reply by tillsteinbach** [reply]
Hmmm, both VWsFriend and Grafana are crashing. What generation of RPI is it?

**Reply by smo85000** [reply]
it's a raspberry pi zero w
I saw IDDataLogger is not compatible with RPI zero W : are you based on IDDataLogger ?

**Reply by tillsteinbach** [reply]
No, but I need at least ARMv7 architecture.
So what might work is:
Zero 2 W
or anything after 2 Mod. B

I have reports that it is working on Gen 3 of the RPI.

**Reply by thgau** [reply]
raspberry pi zero has an arm v6 cpu, most other raspberry models have arm v7 cpu.
There is a similar error described in https://github.com/docker-library/python/issues/565
maybe the mentioned --platform linux/arm/v6 can help...

**Reply by tillsteinbach** [reply]
I don't think it will be possible as the images I rely on are ARMv7 only. I can only suggest to get a slightly newer device.

**Reply by smo85000** [reply]
Thanks for your help. I will try later with another RPI with ARMv7 architecture. See you soon !
